tools/logger/convert.c

```diff
@@ -318,7 +318,7 @@ static double to_usecs(uint64_t time)
 }
 
 /** Justified timestamp width for printf format string */
-static uint8_t timestamp_width(uint8_t precision)
+static int8_t timestamp_width(int8_t precision)
```

Comment:
Please change the `config->precision` but do not change the prototype of `timestamp_width()`. It's not designed to deal with negative numbers at all and every invocation of it makes sure of that.